eventbus/driver/nats.go

```diff
@@ -411,9 +413,30 @@ func (mh *eventSourceMessageHolder) getLastResetTime() int64 {
 }
 
 func (mh *eventSourceMessageHolder) setLastResetTime(t int64) {
-	mh.lock.Lock()
-	defer mh.lock.Unlock()
-	mh.lastResetTime = t
+	{
+		mh.lock.Lock() // since this can be called asyncronously as part of a ConditionReset, we neeed to lock this code
+		defer mh.lock.Unlock()
+		mh.lastResetTime = t
+	}
```

Comment:
Can we share the same lock and simply call `setLastResetTime(0)` instead of `setRestTimeout(0)`? It should be safe to set `lastResetTime` to 0, right?

```
mh.lock.Lock()
defer mh.lock.Unlock()
mh.lastResetTime = t
if t == 0 {
  mh.resetTime = 0
} else {
  mh.resetTime = t + 60
}
```

Reply:
No, one of the changes I made as part of this whole thing is that I never set lastResetTime to 0. lastResetTime now always represents an actual time. This ended up being necessary in a certain scenario.

Reply:
Certain scenario such as?

Reply:
Say you have condition A && B && C,
Say A and B both occur 
Say the pod goes down while the condition reset time occurs
So, we want to be able to ack both A and B when they arrive but we don't: this is because if A arrives, we call reset(), which ends up doing this:
       `if mh.isCleanedUp() {
		mh.setLastResetTime(0)
	}`
That is because the pod went down and B is no longer stored in the eventSourceMessageHolder's memory. 
Then when B arrives we fail to ack it, because we would skip the logic which checks lastResetTime.

Reply:
Good Point!